### DISCUSSION TITLE
Using Hextra on GitHub Page never works

### DISCUSSION THREAD

**Discussion by wlin16** [imfing/hextra#41] [Q&A] [answered]
Hi there, 

Thank you for sharing such as nice theme. 

However, I have been trying to build and deploy Hextra to a GitHub page all night, but I haven't been able to make it work

I've tried using the Hextra template and also tried building from scratch based on Hextra's instruction, but neither approach worked.

On the local server (http://localhost:1313/) I can view the new site using `hugo server --buildDrafts --disableFastRender`. I can also successfully use GitHub Actions to build and deploy it. However, every time I click on the site's link, it only displays the repository's name and the contents of the README.md, not the content the site should have.

I'm not sure if I need to convert the site to static with Hugo. But after trying `hugo` or `hugo -D` in the terminal and then uploading the files onto the GitHub repo, it still just displays the repository's name and the README.md content after clicking the link of the site.

I'm also not sure if the failure is because Hextra requires special flags like `--buildDrafts` and `--disableFastRender `when running `hugo server ` since other themes don't need these flags.

Could you please help me solve this? Or provide guidance on how to correctly deploy a blog with the Hextra theme on GitHub pages? I've been able to deploy other themes based on Hugo to GitHub pages successfully.

**Comment by imfing** [comment]
@wlin16 please start with the [hextra-starter-template](https://github.com/imfing/hextra-starter-template#github-pages) repo. In the README, there's a [Deployment](https://github.com/imfing/hextra-starter-template#deployment) section that tells you how to deploy to GitHub Pages:

> A GitHub Actions workflow is provided in [`.github/workflows/pages.yaml`](https://github.com/imfing/hextra-starter-template/blob/main/.github/workflows/pages.yaml) to [publish to GitHub Pages](https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/) for free. 
We need to [run the workflow manually](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow).

To clarify:

> I'm not sure if I need to convert the site to static with Hugo. But after trying hugo or hugo -D in the terminal and then uploading the files onto the GitHub repo, it still just displays the repository's name and the README.md content after clicking the link of the site.

I recommend using the [GitHub action](https://github.com/imfing/hextra-starter-template/blob/main/.github/workflows/pages.yaml) I provided in the template repo. You don't need to convert the site to static locally using Hugo. 

> I'm also not sure if the failure is because Hextra requires special flags like --buildDrafts and --disableFastRender when running hugo server  since other themes don't need these flags.

No, these flags are not required to build the site. 

> Could you please help me solve this? Or provide guidance on how to correctly deploy a blog with the Hextra theme on GitHub pages? I've been able to deploy other themes based on Hugo to GitHub pages successfully.

- Use the [hextra-starter-template](https://github.com/imfing/hextra-starter-template#github-pages) to create your own repo by clicking the ["Use this template"](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template#creating-a-repository-from-a-template)
- Run the [GitHub action](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow) on your created repo if it's not triggered automatically

After that, you should at least see a site that looks exactly like the template site. And you can start modifying files from there.

**Reply by wlin16** [reply]
Hi, 

Thank you for the reply. 

As mentioned in my previous message, I do have tried using hextra-starter-template to create my own repo by clicking the "Use this template" and have also run the repo with GitHub action successfully. However, there's only showing the name of the repo and the content in README.md, not the template site. 

According to the instruction you sent, I tried to re-create a new repo via using hextra-starter-template and also ensure GitHub Action works fine. However, the template site is still not showing. This is what I've got: https://wlin16.github.io/blog_test/

If it is fine, can I add you on WeChat? My WeChat ID is: Aurum_Au_

**Reply by imfing** [reply, marked as answer]
It was caused by GitHub Pages settings where the deployment source was set to `main` instead of `GitHub Actions`:

![image](https://github.com/imfing/hextra/assets/5097752/18affa8b-fcc1-41ec-93a2-dc05c93768fe)

Set to `GitHub Actions` would fix the problem.

**Reply by whudn** [reply]
sorry it didn't work for me, I already cloned the repository and ran manual github action and changed github pages setting to github action it made no change at all https://github.com/whudn/whudn.github.io

**Reply by imfing** [reply]
@whudn can you try change the baseURL like this: https://github.com/imfing/hextra-starter-template/issues/4#issue-1934100445

**Reply by whudn** [reply]
@imfing thankyou

**Reply by whudn** [reply]
@imfing how about custom domain?

**Reply by imfing** [reply]
> @imfing how about custom domain? 

just change the baseURL accordingly
